### DISCUSSION TITLE
x-for issue with 3rd party library like datatable

### DISCUSSION THREAD

**Discussion by AbedelrahmanD** [alpinejs/alpine#2974] [7. Bugs (archived)] [answered]
When I use x-for inside 3rd part library like datatable and swiper, the data inside x-for doesn't appear, I think the problem is because of the <template> tag, is there a solution? is there a way to make x-for without <template> tag?

Below is the example code, tbody is not showing, but if I put <tr> without x-for, it will appear.
[index.txt](https://github.com/alpinejs/alpine/files/8840820/index.txt)

**Comment by SimoTod** [comment, marked as answer]
When you use a third party library that manipulates the DOM, you should wrap the call into Alpine.mutateDom to pause the mutation observer. Try

```html
<script>
  $(document).ready(function () {
    Alpine.mutateDom(function() {
      $('#example').DataTable()
    }
  }
);
</script>
```
